src/Resources/skeleton/crud/controller/ControllerWithTwig.tpl.php

```diff
@@ -0,0 +1,145 @@
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($<?= $entity_var_singular; ?>);
+            $em->flush();
+
+            return $this->redirectToRoute('<?= $route_name; ?>_edit', ['<?= $entity_identifier; ?>' => $<?= $entity_var_singular; ?>->get<?= ucfirst($entity_identifier); ?>()]);
+        }
+
+        return $this->render('<?= $route_name; ?>/new.html.twig', [
+            '<?= $entity_var_singular; ?>' => $<?= $entity_var_singular; ?>,
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/{<?= $entity_identifier; ?>}/edit", name="edit")
+     * @Method({"GET", "POST"})
+     *
+     * @param Request $request
+     * @param <?= $entity_class_name; ?>  $<?= $entity_var_singular; ?>  The <?= $entity_class_name; ?> entity
+     *
+     * @return Response
+     */
+    public function edit(Request $request, <?= $entity_class_name; ?> $<?= $entity_var_singular; ?>)
+    {
+        $form = $this->createForm(<?= $form_class_name; ?>::class, $<?= $entity_var_singular; ?>);
+        $form->handleRequest($request);
+
+        $deleteForm = $this->createDeleteForm($<?= $entity_var_singular; ?>);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->getDoctrine()->getManager()->flush();
+
+            return $this->redirectToRoute('<?= $route_name; ?>_edit', ['<?= $entity_identifier; ?>' => $<?= $entity_var_singular; ?>->get<?= ucfirst($entity_identifier); ?>()]);
+        }
+
+        return $this->render('<?= $route_name; ?>/edit.html.twig', [
+            '<?= $entity_var_singular; ?>' => $<?= $entity_var_singular; ?>,
+            'form' => $form->createView(),
+            'delete_form' => $deleteForm->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/{<?= $entity_identifier; ?>}", name="delete")
+     * @Method("DELETE")
+     *
+     * @param Request $request
+     * @param <?= $entity_class_name; ?>  $<?= $entity_var_singular; ?>  The <?= $entity_class_name; ?> entity
+     *
+     * @return Response
+     */
+    public function delete(Request $request, <?= $entity_class_name; ?> $<?= $entity_var_singular; ?>)
+    {
+        $form = $this->createDeleteForm($<?= $entity_var_singular; ?>);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $em = $this->getDoctrine()->getManager();
+            $em->remove($<?= $entity_var_singular; ?>);
+            $em->flush();
+        }
+
+        return $this->redirectToRoute('<?= $route_name; ?>_index');
+    }
+
+    /**
+     * Creates a form to delete a <?= $entity_class_name; ?> entity.
+     *
+     * @param <?= $entity_class_name; ?> $<?= $entity_var_singular; ?> The <?= $entity_class_name; ?> entity
+     *
+     * @return \Symfony\Component\Form\FormInterface The form
+     */
+    private function createDeleteForm(<?= $entity_class_name; ?> $<?= $entity_var_singular; ?>)
```

Comment:
Maybe follow the Symfony demo in this case by using a seperate template to generate a delete form? https://github.com/symfony/demo/blob/master/templates/admin/blog/_delete_form.html.twig

Reply:
There is a big question with delete form for me. May be there is a way to create a delete link in order to delete entities from list and more customization in other templates? Delete form is a good and secure way, but we can include it only in edit or show action.

Reply:
With a seperate template this is easily achievable with

    {% include '.../_delete_form.html.twig with {'post': post.id} %}  

but I don't know whether there are security concerns when using multiple forms with the same CSRF token.